Telecoms Resprite

Spangs04 opened the pull request:
<!-- Guidelines: https://docs.spacestation14.io/en/getting-started/pr-guideline -->

## About the PR
<!-- What did you change? -->
Resprited Telecommunications Servers (and changed a few sprite states for some existing servers)
## Why / Balance
<!-- Discuss how this would affect game balance or explain why it was changed. Link any relevant discussions or issues. -->
The current server sprites that are in the game look a little boring, and I feel that they should have a little bit more of intractability. I changed the sprites for the telecoms and added indicators for the server's key(s) to give it more visual clarity.
## Technical details
<!-- Summary of code changes for easier review. -->
YML and JSON
## Media
<!-- Attach media if the PR makes ingame changes (clothing, items, features, etc). 
Small fixes/refactors are exempt. Media may be used in SS14 progress reports with credit. -->

![image_2025-05-03_141945908](https://github.com/user-attachments/assets/0a10956d-94c0-4b1f-ae40-60925e96f6fa)
(stripes are changed to be slightly less dark now need to update media)
## Requirements
<!-- Confirm the following by placing an X in the brackets [X]: -->
- [X] I have read and am following the [Pull Request and Changelog Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html).
- [X] I have added media to this PR or it does not require an ingame showcase.
<!-- You should understand that not following the above may get your PR closed at maintainer’s discretion -->

## Breaking changes
<!-- List any breaking changes, including namespaces, public class/method/field changes, prototype renames; and provide instructions for fixing them.
This will be posted in #codebase-changes. -->
N/A

**Changelog**
<!-- Add a Changelog entry to make players aware of new features or changes that could affect gameplay.
Make sure to read the guidelines and take this Changelog template out of the comment block in order for it to show up.
Changelog must have a :cl: symbol, so the bot recognizes the changes and adds them to the game's changelog. -->
<!--
-->🆑 
- tweak: Resprited Telecomms

kosticia commented:
Sprite looks cool, and your changelog missing :cl: symbol

UbaserB commented:
i like it a lot the stripes just need to be tweaked it looks like weirdly plastic idk

Spangs04 commented:
> i like it a lot the stripes just need to be tweaked it looks like weirdly plastic idk
> 
> 
> 
> actually the whole thing kinda looks a little too plastic idk

Im fine to change it up if needed, the current color scheme is actually directly from the old servers and the stripes/frame they use. I am always open to feedback about them though so thanks for the reply!

iaada commented:
Would you be up to having the telecom server change sprites depending on if a com key is inside it or not? Something like the bit in the top right turns off if it's not "broadcasting" anything. It would be nice to help identify comms sabotage.

LaCumbiaDelCoronavirus commented:
New sprites look amazing, however I have 2 things to say about it
- Not really an issue but the servers themselves (the left half of each rack) look a bit flat (..which confused me at first), maybe make them slightly darker/lighter than the rest of the rack (since they're both currently the same color) to show depth?
- Camera router and comms server look too similar other than slightly recoloured stripes

Spangs04 commented:
> New sprites look amazing, however I have 2 things to say about it
> 
> - Not really an issue but the servers themselves (the left half of each rack) look a bit flat (..which confused me at first), maybe make them slightly darker/lighter than the rest of the rack (since they're both currently the same color) to show depth?
> 

Yeah, I'll totally make them lighter. I thought I made em lighter before but It seems that I did not realize that they didn't update

> - Camera router and comms server look too similar other than slightly recoloured stripes
> 
> 

I agree about that, I am currently working on a better stripe for the both of them already and they should look a lot better (soonish). I will update the media to reflect the changes once they are made as well

ArcaneOcto commented:
This looks so cool. I agree that it would be even better if it displayed the encryption key inside it. Even if it only shows the first/last key inserted.

Spangs04 commented:
I am very close to being done with the massive overhaul of the telecoms displays and how the sprites are rendered. Just need to figure a few things out, and it should be done soon.

Spangs04 commented:
Ok, I got the sprites for the telecoms completed, but I am stuck on how to display them properly depending on what keys/key is inserted. The key-specific sprites work on the prefab telecoms, but I may need assistance when a player inserts the key/keys into the server to change it accordingly.

iaada commented:
Best way I can think to do it is with `ItemMapper`, same way that the crayon box or belts do it. Though it would involve adding a tag for each type of encryption key, and I'm not sure how the ordering would work if multiple sprites were in the same place i.e. multiple keys in the same server.

If you want to go simpler you can use `ItemMapper` to add a generic icon for if any encryption key exists.

Spangs04 commented:
I have finished with all of the servers, telecoms, and cameras. I have also implemented a visual for the key in the telecom server! Everything is complete and working.

iaada commented on Resources/Prototypes/Entities/Structures/Machines/telecomms.yml:
```diff
@@ -8,23 +8,139 @@
+        minCount: 1
+      tele-server-on-multiple:
+        whitelist:
+          tags:
```
Instead of a whitelist checking for each individual tag, it's probably safer for a whitelist to check for `EncryptionKey` component here.

Spangs04 replied:
Completely fair and would be much simpler. Thank you for taking a look at it.

Spangs04 commented:
I changed how the telecoms display the keys to be more generic and have less bloat. Slightly changed the sprite for the server having no key inside of it as well.

Spangs04 commented:
Test fail related to #35933

iaada commented on Resources/Prototypes/Entities/Structures/Machines/telecomms.yml:
```diff
@@ -8,23 +8,138 @@
+        whitelist:
+          tags:
+          - EncryptionCommand
+      tele-server-on-default:
```
If you move this layer before the departments in `SpriteComponent`, you can do away with the `EncryptionElse` tag. I'm scared by the reliance on layer order but it works and I don't know if I'm supposed to be scared, anyway.
```suggestion
      tele-server-on-default:
        whitelist:
          components:
          - EncryptionKey
        minCount: 1
```

Spangs04 replied:
Agreed, better implimentation

iaada commented on Resources/Prototypes/Entities/Structures/Machines/telecomms.yml:
```diff
@@ -8,23 +8,138 @@
+        minCount: 1
+      tele-server-on-multiple:
+        whitelist:
+          tags:
```
Instead of creating a new tag, look for the component instead.
```suggestion
          components:
```

iaada commented on Resources/Prototypes/Entities/Structures/Machines/telecomms.yml:
```diff
@@ -8,23 +8,138 @@
+        minCount: 2
+      tele-server-no-key:
+        whitelist:
+          tags:
```
Replace tag with component again, and move this map up to the top of `ItemMapper` to reflect its order in `SpriteComponent`.

iaada commented on Resources/Prototypes/Entities/Structures/Machines/telecomms.yml:
```diff
@@ -8,23 +8,138 @@
           False: { visible: false }
+        tele-server-on-common:
+          True: { shader: unshaded }
+          False: { shader: shaded }
```
My personal preference is to make these invisible when the power is off, particularly because this section gets turned off when the cam router gets turned off.

Spangs04 replied:
One issue I have encountered is that the visible layer of things in the item mapper cannot have its visibility toggled on and off. For some reason, it is not affected by it, but it is affected by things such as shaders in the visualizer. Is there a different visibility for the item mapped sprites?

walksanatora commented:
I just realized why I dont like these sprites. ["The future never looked soo old"](https://docs.spacestation14.com/en/space-station-14/art.html#the-future-never-looked-so-old) and this looks a bit too forward for this.

EmoGarbage404 reviewed (dismissed):
So i do like the premise but i think it's being applied a bit too broadly here.

I'd keep the RD, crew monitoring, and camera servers as they are. I think the basic server box is a lot better for something you're not meant to directly interface with.

I do like this design concept a lot more for telecomms. However, this current iteration is a little bit too clean IMO.

![hub_anim_South](https://github.com/user-attachments/assets/0df1c5a5-1940-4634-837f-3b665c581387)

It might be best to take inspiration from some older SS13 sprites and go for a darker frame and a bit more grungy (since this is a technical sort of machine).

I love the animations and job indicators, though. Those should definitely stay.

Spangs04 commented:
> So i do like the premise but i think it's being applied a bit too broadly here.
> 
> I'd keep the RD, crew monitoring, and camera servers as they are. I think the basic server box is a lot better for something you're not meant to directly interface with.
Totally, I can swing this more towards telecoms if that's what is preferred
> 
> I do like this design concept a lot more for telecomms. However, this current iteration is a little bit too clean IMO.
I can totally make it more grungy ill be sure to improve it
> 
> ![hub_anim_South](https://private-user-images.githubusercontent.com/98561806/438398470-0df1c5a5-1940-4634-837f-3b665c581387.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDU4OTEzNjcsIm5iZiI6MTc0NTg5MTA2NywicGF0aCI6Ii85ODU2MTgwNi80MzgzOTg0NzAtMGRmMWM1YTUtMTk0MC00NjM0LTgzN2YtM2I2NjVjNTgxMzg3LmdpZj9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA0MjklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNDI5VDAxNDQyN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWY5M2JmNjlmMWMyZDcwZGM0ODJiNzEyNjFjMTA1YWQzY2M1ZTU5ZDg5NmQ4NWU5NzNlN2I5Y2Q3NGZmMzBiMDUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.cYt9GUA9mF5bJETejCJnMpbIjiTksbStbwbC4-pcEMI) [ ![hub_anim_South](https://private-user-images.githubusercontent.com/98561806/438398470-0df1c5a5-1940-4634-837f-3b665c581387.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDU4OTEzNjcsIm5iZiI6MTc0NTg5MTA2NywicGF0aCI6Ii85ODU2MTgwNi80MzgzOTg0NzAtMGRmMWM1YTUtMTk0MC00NjM0LTgzN2YtM2I2NjVjNTgxMzg3LmdpZj9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA0MjklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNDI5VDAxNDQyN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWY5M2JmNjlmMWMyZDcwZGM0ODJiNzEyNjFjMTA1YWQzY2M1ZTU5ZDg5NmQ4NWU5NzNlN2I5Y2Q3NGZmMzBiMDUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.cYt9GUA9mF5bJETejCJnMpbIjiTksbStbwbC4-pcEMI) ](https://private-user-images.githubusercontent.com/98561806/438398470-0df1c5a5-1940-4634-837f-3b665c581387.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDU4OTEzNjcsIm5iZiI6MTc0NTg5MTA2NywicGF0aCI6Ii85ODU2MTgwNi80MzgzOTg0NzAtMGRmMWM1YTUtMTk0MC00NjM0LTgzN2YtM2I2NjVjNTgxMzg3LmdpZj9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA0MjklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNDI5VDAxNDQyN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWY5M2JmNjlmMWMyZDcwZGM0ODJiNzEyNjFjMTA1YWQzY2M1ZTU5ZDg5NmQ4NWU5NzNlN2I5Y2Q3NGZmMzBiMDUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.cYt9GUA9mF5bJETejCJnMpbIjiTksbStbwbC4-pcEMI) [ ](https://private-user-images.githubusercontent.com/98561806/438398470-0df1c5a5-1940-4634-837f-3b665c581387.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDU4OTEzNjcsIm5iZiI6MTc0NTg5MTA2NywicGF0aCI6Ii85ODU2MTgwNi80MzgzOTg0NzAtMGRmMWM1YTUtMTk0MC00NjM0LTgzN2YtM2I2NjVjNTgxMzg3LmdpZj9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA0MjklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNDI5VDAxNDQyN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWY5M2JmNjlmMWMyZDcwZGM0ODJiNzEyNjFjMTA1YWQzY2M1ZTU5ZDg5NmQ4NWU5NzNlN2I5Y2Q3NGZmMzBiMDUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.cYt9GUA9mF5bJETejCJnMpbIjiTksbStbwbC4-pcEMI)
> 
> It might be best to take inspiration from some older SS13 sprites and go for a darker frame and a bit more grungy (since this is a technical sort of machine).
> 
> I love the animations and job indicators, though. Those should definitely stay.

Spangs04 commented:
Ok, I have changed the files for the science, suit coords, and camera server sprites back to their old states with some slight changes I made to at least update them slightly for some things (like audio not disabling when power goes out or the indicators that existed in the files already). Also cleaned a lot of my old code for the most part, and am now working on the updated sprites for telecoms as requested.

Spangs04 commented:
Made the servers darker and tried to maintain the old color scheme of the original telecom servers

Spangs04 commented:
Feel free to offer any feedback/ideas on how I could improve, as I feel that there is always room for it

Spangs04 commented:
I'm not sure if that test fail is related, but I could be wrong
